### DISCUSSION TITLE
RFC #1 - Package Name

### DISCUSSION THREAD

**Discussion by RaddishIoW** [HaylLtd/cordova-plugin-background-geolocation#7] [General]
## Proposal
In order to differentiate this fork of the original project and help with visibility, it is proposed to rename the package, preferably with an “unscoped” NPM name (e.g. “cordova-background-geolocation”, rather than “@haylltd/cordova-background-geolocation”).

Suggestions for a new name are being accepted below.

### Discussion Points
* Name should ideally include “cordova”, “geolocation” and “background”, to clearly indicate the purpose of the package.
* If an unscoped package name cannot be secured due to NPM package naming conflicts, then a scope should be considered, but one which can be maintained by a community rather than tied to an individual or company.
* The repository name should be changed if possible to reflect this new package name, to avoid confusion.

**Comment by RaddishIoW** [comment]
This utility is useful for finding out if a name is taken: https://remarkablemark.org/npm-package-name-checker/

**Comment by RaddishIoW** [comment]
`cordova-bg-geolocation` is available at the moment, and would be a good option I think.

**Reply by HarelM** [reply]
Most Cordova plugin names end with the "plugin" postfix, I suggest the following:
`cordova-background-geolocation-plugin`
or 
`cordova-bg-gelocation-plugin`
I prefer the first (the longer one).

**Reply by RaddishIoW** [reply]
Yep - `cordova-background-geolocation-plugin` works for me. Nice and descriptive, and it's available according to the name checker. :)
I'll open an issue to get the name changed.

**Reply by HarelM** [reply]
I'm not sure why I state that most Cordova plugin names end with the "plugin" postfix, I guess I wan't very sharp when I wrote it, as most plugins start with cordova-plugin-bla-bla-bla...
Since we haven't published anything officially but only release candidates we can still change it if we want.
I'm not sure I have a better name though, so your call...

**Reply by RaddishIoW** [reply]
My vote is to leave it as it is now, as it's descriptive and has all the relevant parts in the name. :)

**Reply by HarelM** [reply]
Sure. No problem.
